monkey/monkey_island/cc/server_utils/encryption/data_store_encryptor.py

```diff
@@ -1,49 +1,29 @@
-import os
+from __future__ import annotations
 
 # PyCrypto is deprecated, but we use pycryptodome, which uses the exact same imports but
 # is maintained.
-from Crypto import Random  # noqa: DUO133  # nosec: B413
+from typing import Union
 
 from monkey_island.cc.server_utils.encryption import KeyBasedEncryptor
-from monkey_island.cc.server_utils.file_utils import open_new_securely_permissioned_file
 
-_encryptor = None
+_encryptor: Union[None, DataStoreEncryptor] = None
 
 
 class DataStoreEncryptor:
-    _BLOCK_SIZE = 32
-    _KEY_FILENAME = "mongo_key.bin"
-
-    def __init__(self, key_file_dir):
-        key_file = os.path.join(key_file_dir, self._KEY_FILENAME)
-
-        if os.path.exists(key_file):
-            self._load_existing_key(key_file)
-        else:
-            self._init_key(key_file)
-
-        self._key_base_encryptor = KeyBasedEncryptor(self._cipher_key)
-
-    def _init_key(self, password_file_path: str):
-        self._cipher_key = Random.new().read(self._BLOCK_SIZE)
-        with open_new_securely_permissioned_file(password_file_path, "wb") as f:
-            f.write(self._cipher_key)
-
-    def _load_existing_key(self, key_file):
-        with open(key_file, "rb") as f:
-            self._cipher_key = f.read()
+    def __init__(self, key_based_encryptor: KeyBasedEncryptor):
+        self._key_based_encryptor = key_based_encryptor
```

Comment:
If the DataStoreEncryptor is a thin wrapper around KeyBasedEncryptor with no logic, I'd rather `get_datastore_encryptor()` return an IEncryptor and just return a concrete KeyBasedEncryptor. We can refactor everything to use `encrypt()` and `decrypt()` instead of `enc()` and `dec()`, which is better anyway.